FEM-2673 Add more analytics events properties

giladna opened the pull request:
Make OKHTTP the default http requests library

Added 2 new events 
        CONNECTION_ACQUIRED,
        OUTPUT_BUFFER_COUNT_UPDATE

giladna committed:
Kava Reposts Enhancements

giladna committed:
add event OutputBufferCountUpdate

giladna committed:
protect listener from null

giladna committed:
use lower case

giladna committed:
add bufferPosition to playhead event

giladna committed:
to lower case

noamtamim commented on playkit/src/main/java/com/kaltura/playkit/Utils.java:
```diff
@@ -203,4 +213,48 @@ public static boolean equals(Object a, Object b) {
         return String.format(Locale.ROOT, format, args);
     }
+
+    public static String getNetworkClass(Context context) {
```
Some of the constants are only available in latest SDKs, won't it cause a crash?

giladna replied:
tested this code with 4.1.2 device do you think it is enough?

giladna replied:
tested and seems fine you cannot print it but it is fine in run time

noamtamim commented on playkit/src/main/java/com/kaltura/playkit/player/ExoAnalyticsAggregator.java:
```diff
@@ -52,5 +84,55 @@ public void onLoadError(EventTime eventTime, MediaSourceEventListener.LoadEventI
+//                //log.v("onLoadStarted ignore url = " + loadedURL);
+//            }
+//        }
+//    }
```
Remove....

noamtamim commented on playkit/src/main/java/com/kaltura/playkit/player/ExoAnalyticsAggregator.java:
```diff
@@ -1,20 +1,39 @@
+class ExoAnalyticsAggregator extends EventListener implements AnalyticsListener {
 
+    private static final PKLog log = PKLog.get("ExoAnalyticsAggregator");
+    private final Map<String, Long>  domainCallStartRelTimeMap = new ConcurrentHashMap<>();
```
Why "domain"? Isn't it a map of URLs?

noamtamim commented on playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java:
```diff
@@ -406,9 +406,13 @@ private MediaSource buildExternalSubtitleSource(int subtitleId, PKExternalSubtit
+                    builder.eventListenerFactory(okListenerFactory);
+                }
+            } else {
+                builder.eventListener(analyticsAggregator);
```
Please check if there can be more than one listener -- then we will be able to have both profiler and analytics aggregator.

giladna replied:
move the  builder.eventListener(analyticsAggregator); to be first without else

noamtamim commented on playkit/src/main/java/com/kaltura/playkit/PlayerEvent.java:
```diff
@@ -253,13 +257,63 @@ public String toString() {
 
+    public static class ConnectionAcquired extends PlayerEvent {
+
+        public final long connectDurationMs;
```
Let's try to keep 3 values here, if it's not too complicated: DNS duration, connect duration, TLS duration.

giladna replied:
combine all to one object and updated the whole chain

giladna committed:
code review changes
fix user agent to handle mobile + tablet + tv
add dns and tls data to CONNECTION_ACQUIRED event
